hbase-server/pom.xml

```diff
@@ -437,24 +438,83 @@
           <skipAssembly>true</skipAssembly>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>add-test-source</id>
+            <goals>
+              <goal>add-test-resource</goal>
+            </goals>
+            <phase>generate-test-sources</phase>
+            <configuration>
+              <!-- Add the hbase-webapps directory to the test resources -->
+              <resources>
+                <resource>
+                  <!-- Directory containing hbase-webapps -->
+                  <directory>target/${hbase.webapps.dir}</directory>
+                  <!-- Target directory under test-classes -->
+                  <targetPath>${hbase.webapps.dir}</targetPath>
```

Comment:
Do we need the target directory because we compile JSPs at compile time ?

Reply:
yes we need targetPath, otherwise the files would be copied directly into target/test-classes/master but we want to have target/test-classes/hbase-webapps/master

Reply:
I meant the source directory: We cannot ass the source directory directly, because we need the .JSP files compiled first, right ?

Reply:
> I meant the source directory: We cannot add the source directory directly, because we need the .JSP files compiled first, right ?

Yes right source directory cannot be added as they need to be compiled.